src/twilio-sms-notification-sink.ts

```diff
@@ -0,0 +1,47 @@
+import { Notification, ResourceId } from './data-model';
+import { NotificationSink } from './ports';
+import { Twilio } from 'twilio';
+
+/**
+ * Sms notification
+ */
+export interface SmsNotification extends Notification {
+  body: string;
+}
+
+export type SmsNumber = string;
+
+export type ResourceSms = {
+  resourceId: ResourceId;
+  smsNumber: SmsNumber;
+};
+
+export interface ResourceSmsNumberRepository {
+  findBy(resourceIds: ResourceId[]): Promise<ResourceSms[]>;
+}
+
+export class TwilioSmsNotificationSink
+  implements NotificationSink<SmsNotification>
+{
+  private twilio: Twilio;
+  constructor(
+    private readonly twilioAccount: { username: string, password: string },
+    private readonly senderSmsNumber: string,
+    private readonly resourceIdToReceiverSmsNumberMapper: ResourceSmsNumberRepository,
+  ) {
+    this.twilio = new Twilio(twilioAccount.username, twilioAccount.password);
+  }
+
+  async push(notification: SmsNotification, recipients: ResourceId[]) {
+    const recipientSmSNumbers = await this.resourceIdToReceiverSmsNumberMapper.findBy(
+      recipients,
+    );
+    return Promise.allSettled(recipientSmSNumbers.map(({ smsNumber }) => {
```

Comment:
Let's log failed sends like in dialect sink?  Would be great if you add the same to email sink, should be a copy paste
wdyt?

Reply:
Agreed. I will add in both places.

Reply:
ah finally addressed this comment in latest commit.